R/finalize_explanation.R

```diff
@@ -153,3 +168,238 @@ compute_shapley_new <- function(internal, dt_vS) {
+#' compute_MSEv_eval_crit(
+#'   internal = explanation$internal,
+#'   processed_vS_list = explanation$internal$output,
+#'   exclude_empty_grand_coalitions = FALSE,
+#'   return_as_dt = TRUE
+#' )
+#'
+#' # Exclude the empty and grand coalitions from the computations and return the result as a list
+#' compute_MSEv_eval_crit(
+#'   internal = explanation$internal,
+#'   processed_vS_list = explanation$internal$output,
+#'   exclude_empty_grand_coalitions = TRUE,
+#'   return_as_dt = FALSE
+#' )
+#'
+#' # Include the empty and grand coalitions from the computations and return the result as a list
+#' compute_MSEv_eval_crit(
+#'   internal = explanation$internal,
+#'   processed_vS_list = explanation$internal$output,
+#'   exclude_empty_grand_coalitions = FALSE,
+#'   return_as_dt = FALSE
+#' )
+#'
+#' @author Lars Henry Berge Olsen
+#' @export
+compute_MSEv_eval_crit <- function(internal,
+                                   processed_vS_list,
+                                   p = shapr:::get_p(processed_vS_list$dt_vS, internal),
+                                   exclude_empty_grand_coalitions = TRUE,
+                                   use_shapley_weights = FALSE,
+                                   return_as_dt = TRUE) {
+
+  # Get the number of observations to explain
+  n_explain <- internal$parameters$n_explain
+
+  # Get the number of unique coalitions, where two of them are the empty and full set.
+  # This is 2^M if internal$parameters$exact is TRUE or some value below 2^M if sampled version.
+  n_combinations <- internal$parameters$n_combinations
+
+  # Get the relevant combinations, i.e., if we are to remove the empty and grand coalitions,
+  # which are identical for all methods and thus is not effected by the used method.
+  id_combination_numbers <- if (exclude_empty_grand_coalitions) seq(2, n_combinations - 1) else seq(1, n_combinations)
+
+  # Get the estimated contribution functions for the different coalitions and explicands
+  vS <- as.matrix(processed_vS_list$dt_vS[id_combination_numbers, -"id_combination"])
+
+  # Square the difference between the predicted response f(x) and estimated contribution
+  # function v(S) = E_{\hat{p}(X_sbar | X_s = x_s)} [f(X_sbar, x_s) | X_s = x_s)].
+  dt_squared_diff_original <- sweep(vS, 2, p)^2
+
+  # Check if we are doing weighted mean with the Shapley kernel weights as the weights or using uniform weights
+  if (use_shapley_weights) {
+    if (!exclude_empty_grand_coalitions) {
+      warning(paste0("We do not reccomend seting `exclude_empty_grand_coalitions = FALSE` when ",
+                     "`use_shapley_weights = TRUE`, as the weights of the empty and full coalitions ",
+                     "drastically outweights the other coalitions. This causes the MSEv criterion to be ",
+                     "more or less identical for all approaches."))
+    }
+
+    # We are using the weighted
+    averaging_weights = internal$objects$X$shapley_weight
```

Comment:
I guess you can simplify the code by defining averaring_weights conditional on use_shapley_weights in one line, to then reuse the code in lines 340-345 for both case.